ogr/services/github/service.py

```diff
@@ -47,64 +56,49 @@ def __init__(
         github_app_id: str = None,
         github_app_private_key: str = None,
         github_app_private_key_path: str = None,
+        tokman_instance_url: str = None,
+        github_authentication: GithubAuthentication = None,
         **_,
     ):
+        """
+        If multiple authentication methods are provided, they are prioritised:
+            1. Tokman
+            2. GithubApp
+            3. Token (which is also default one, that works without specified token)
```

Comment:
I might have missed the discussion: could this be made explicit, instead of computing it, in order to remove some complexity?

Reply:
We are loading the services from config (yaml) where you can easily set those arguments. (The authenticator instance is possible as well.)

Reply:
I'm not sure if there is *better* solution than either of those...

- So far it has been handled in `GithubService` silently prioritizing GitHub App
  _coughing_ even with side effect that defaulted to token if no key given and empty file provided _coughing_ (that also presented an interesting issue with tests failing for invalid key format in file...)
- If it is handled explicitly at initialization, seems like a mess too
- Current implementation allows keeping the logic inside those authentication classes with a little cost in service
---
- Checking if provided *exactly* one way to authenticate raises some complications because of the 3 attributes for GitHub App
  which then results in _creative_ dictionary that is used to resolve it later or explicit if-statements in service class...
- We could deprecate current interface and enforce passing authentication object in parameter, however that would mean problem in every place the GithubService is constructed and also (for the purpose of deprecation) would still keep the mess inside the service class
  - we would have to take care of this in `ogr/factory.py`, so it seems more like moving problem from one place to another for me?
  - if created from one of the factory functions, it shouldn't be that big of a deal, cause it could be fixed there; however explicitly using `GithubService` constructor...

Reply:
I see, thanks for the explanation!

Given all this, I think it's fine to leave it as it is for now.